### PR TITLE
ED-1415 add confirm company view

### PR DESCRIPTION
**Pull request by sdonk**

### CODE REVIEW

File: enrolment/views.py [hunk 1]
```diff
@@ -159,20 +95,59 @@ def dispatch(self, request, *args, **kwargs):
...
+    @staticmethod
+    def store_companies_house_profile_in_session_and_validate(
+            session, company_number
+    ):
+        try:
+            helpers.store_companies_house_profile_in_session(
+                session=session,
+                company_number=company_number,
+            )
+        except requests.exceptions.RequestException as error:
```
Comment (konradko) [inline enrolment/views.py]: this actually should be `HttpError`

File: enrolment/views.py [hunk 2]
```diff
@@ -198,9 +177,12 @@ def done(self, *args, **kwags):
         return TemplateResponse(self.request, template)
 
     def get_context_data(self, *args, **kwargs):
-        return super().get_context_data(
+        ctx = super().get_context_data(
             form_labels=self.form_labels, *args, **kwargs
         )
+        if self.storage.current_step == self.COMPANY:
```
Comment (konradko) [inline enrolment/views.py]: not sure why this got here

File: enrolment/forms.py [hunk 3]
```diff
@@ -211,15 +135,19 @@ def serialize_international_buyer_forms(cleaned_data):
...
     @returns dict
 
     """
 
     return {
-        'name': name,
+        'company_name': data['company_name'],
+        'company_number': data['company_number'],
+        'company_address': ", ".join(
+            data['registered_office_address'].values()
```
Comment (konradko) [inline enrolment/forms.py]: this gives a smell

File: enrolment/helpers.py [hunk 4]
```diff
@@ -20,14 +19,20 @@
...
+
+
+def store_companies_house_profile_in_session(session, company_number):
+    details = get_company_from_companies_house(company_number)
     session[COMPANIES_HOUSE_PROFILE_SESSION_KEY] = {
         'company_name': details['company_name'],
         'company_status': details['company_status'],
         'date_of_creation': details['date_of_creation'],
+        'company_number': company_number,
+        'registered_office_address': details['registered_office_address']
```
Comment (konradko) [inline enrolment/helpers.py]: maybe the join should be here, or even more explicit formatting using all fields returned

File: enrolment/templates/company-form.html [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {% extends "form-wrapper.html" %}
 {% block form_title %}Create your company’s profile{% endblock%}
```
Comment (konradko) [inline enrolment/templates/company-form.html]: maybe this should say 'confirm your company's details'

File: enrolment/tests/test_forms.py [hunk 6]
```diff
@@ -53,7 +48,7 @@ def test_indent_invalid_mixin_installed():
 
 @patch.object(validators, 'company_unique', Mock())
 def test_company_form_rejects_missing_data():
```
Comment (konradko) [inline enrolment/tests/test_forms.py]: don't think this test is longer valid

File: enrolment/tests/test_forms.py [hunk 7]
```diff
@@ -66,103 +61,6 @@ def test_company_form_fields():
...
-    assert form.is_valid() is False
-
-    form.errors['company_number'] == [forms.MESSAGE_TRY_AGAIN_LATER]
-
-
-@patch.object(helpers, 'store_companies_house_profile_in_session', Mock())
-@patch.object(helpers, 'get_company_status_from_session',
-              Mock(return_value='active'))
-@patch.object(validators, 'company_active')
-def test_company_form_handles_company_active_validation(
```
Comment (konradko) [inline enrolment/tests/test_forms.py]: we need this test that will be checking both on landing page and /register/company

File: enrolment/views.py [hunk 8]
```diff
@@ -159,20 +95,59 @@ def dispatch(self, request, *args, **kwargs):
...
+                )
+            else:
+                raise ValidationError(
+                    'Error. Please try again later.'
+                )
+        else:
+            company_status = helpers.get_company_status_from_session(
+                session
+            )
+            validators.company_active(company_status)
```
Comment (konradko) [inline enrolment/views.py]: this needs to be on landing page lookup too

File: enrolment/helpers.py (hunk 4 above)
Comment (sdonk) [inline enrolment/helpers.py]: This is not where you are displaying address, I wouldn't join here.

File: enrolment/tests/test_forms.py (hunk 6 above)
Comment (sdonk) [inline enrolment/tests/test_forms.py]: Why doesn't it fail then?

File: enrolment/templates/company-form.html (hunk 5 above)
Comment (sdonk) [inline enrolment/templates/company-form.html]: We should use the template called company-form-confirm for this step, are we not?

Comment (konradko) [inline enrolment/templates/company-form.html]: not really, the form / step is called just `company`

File: enrolment/tests/test_forms.py [hunk 9]
```diff
@@ -216,10 +112,16 @@ def test_serialize_international_buyer_forms():
...
+            'company_name': 'Example',
+            'company_number': 1234,
+            'registered_office_address': {'line1': 'foo', 'line2': 'bar'}
+        }
     )
     expected = {
-        'name': 'Example'
+        'company_name': 'Example',
+        'company_number': 1234,
+        'company_address': 'foo, bar'
```
Comment (sdonk) [inline enrolment/tests/test_forms.py]: this test will randomly fail because a `dict` is not ordered and the join can merge the strings in any order

Comment (konradko) [inline enrolment/tests/test_forms.py]: true

Comment (konradko) [inline enrolment/tests/test_forms.py]: the join needs to be replaced with formatting based on the response fields - will do

File: enrolment/tests/test_views.py [hunk 10]
```diff
@@ -431,6 +367,108 @@ def test_landing_page_submit_valid_form_redirects(
...
+
+    with patch('enrolment.validators.company_unique') as mock:
+        mock.side_effect = ValidationError('Company already exists')
+
+        response = client.get(
+            reverse('register', kwargs={'step': 'company'}),
+            {'company_number': 12345678}
+        )
+
+    assert 'Company already exists' in str(response.content)
```
Comment (konradko) [inline enrolment/tests/test_views.py]: 'already registered' more accurate

**Pull request merged**